manifests/init.pp

```diff
@@ -208,65 +213,77 @@
 #   Disable com2sec, group, and access in snmpd.conf
 #
 class snmp (
-  Enum['present','absent'] $ensure = $snmp::params::ensure,
-  $agentaddress                 = $snmp::params::agentaddress,
-  Array[String[1]] $snmptrapdaddr = $snmp::params::snmptrapdaddr,
-  $ro_community                 = $snmp::params::ro_community,
-  $ro_community6                = $snmp::params::ro_community6,
-  $rw_community                 = $snmp::params::rw_community,
-  $rw_community6                = $snmp::params::rw_community6,
-  $ro_network                   = $snmp::params::ro_network,
-  $ro_network6                  = $snmp::params::ro_network6,
-  $rw_network                   = $snmp::params::rw_network,
-  $rw_network6                  = $snmp::params::rw_network6,
-  $contact                      = $snmp::params::contact,
-  $location                     = $snmp::params::location,
-  $sysname                      = $snmp::params::sysname,
-  $services                     = $snmp::params::services,
-  Array[String[1]] $com2sec     = $snmp::params::com2sec,
-  Array[String[1]] $com2sec6    = $snmp::params::com2sec6,
-  Array[String[1]] $groups      = $snmp::params::groups,
-  Array[String[1]] $views       = $snmp::params::views,
-  Array[String[1]] $accesses    = $snmp::params::accesses,
-  Array[String[1]] $dlmod       = $snmp::params::dlmod,
-  Array[String[1]] $extends     = $snmp::params::extends,
-  Array[String] $snmpd_config   = $snmp::params::snmpd_config,
+  # Usage
```

Comment:
re-ordering everything might make logic sense, though it makes the PR hard to merge because of all the superficial changes.